x-pack/platform/packages/shared/kbn-inference-cli/src/eis/get_service_configuration.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { promises as Fs } from 'fs';
+import Path from 'path';
+import os from 'os';
+import simpleGit from 'simple-git';
+import { load } from 'js-yaml';
+
+class GitCheckoutError extends Error {
+  constructor(cause: Error) {
+    super(`Failed to checkout repository. Make sure you've authenticated to Git`, { cause });
+  }
+}
+
+async function getFiles(files: string[]): Promise<string[]> {
+  // Create a temporary directory
+  const tmpDir = await Fs.mkdtemp(Path.join(os.tmpdir(), 'serverless-gitops-'));
+  const git = simpleGit(tmpDir);
+
+  // Initialize an empty repository and add remote
+  await git.init();
+  await git.raw(['config', 'core.sparseCheckout', 'true']);
+
+  const sparseCheckoutPath = Path.join(tmpDir, '.git', 'info', 'sparse-checkout');
+  await Fs.writeFile(sparseCheckoutPath, files.join('\n'), 'utf-8');
+
+  async function pull() {
+    return await git.pull('origin', 'main', { '--depth': '1' });
+  }
+
+  await git.addRemote('origin', `git@github.com:elastic/serverless-gitops.gitf`);
```

Comment:
I think there is a typo 
```suggestion
  await git.addRemote('origin', `git@github.com:elastic/serverless-gitops.git`);
```

Reply:
yes lol, thank you! was using this to test a failure state